docs/settings.md
# All Django Settings options

| Setting                                          | Description                                                                                                                                                                        |
|--------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `MICROSOFT_SSO_ADMIN_ENABLED`                    | Enable SSO only when allowed on Admin pages. Default: `None`                                                                                                                       |
| `MICROSOFT_SSO_ALLOWABLE_DOMAINS`                | List of domains that will be allowed to create users. Default: `[]`                                                                                                                |
| `MICROSOFT_SSO_ALWAYS_UPDATE_USER_DATA`          | If true, update default user info from Microsoft data at every login. This will also make their password unusable. Otherwise, all of this happens only on create. Default: `False` |
| `MICROSOFT_SSO_APPLICATION_ID`                   | The Microsoft OAuth 2.0 Web Application Client ID. Default: `None`                                                                                                                 |
| `MICROSOFT_SSO_AUTHENTICATION_BACKEND`           | The authentication backend to use.  Default: `None`                                                                                                                                |
| `MICROSOFT_SSO_AUTHORITY`                        | A info that defines the token authority. You should set it with your tenant URL or AuthorityBuilder instance. Default: `None`                                                      |
| `MICROSOFT_SSO_AUTO_CREATE_FIRST_SUPERUSER`      | If True, the first user that logs in will be created as superuser if no superuser exists in the database at all. Default: `False`                                                  |
| `MICROSOFT_SSO_AUTO_CREATE_USERS`                | Enable or disable the auto-create users feature. Default: `True`                                                                                                                   |
| `MICROSOFT_SSO_CACHE_NAME`                       | The name of the Django cache to use for storing tokens. Default: `default`                                                                                                         |
| `MICROSOFT_SSO_CALLBACK_DOMAIN`                  | The netloc to be used on Callback URI. Default: `None`                                                                                                                             |
| `MICROSOFT_SSO_CLIENT_SECRET`                    | The Microsoft OAuth 2.0 Web Application Client Secret. Default: `None`                                                                                                             |
| `MICROSOFT_SSO_ENABLE_LOGS`                      | Show Logs from the library. Default: `True`                                                                                                                                        |
| `MICROSOFT_SSO_ENABLE_MESSAGES`                  | Show Messages using Django Messages Framework. Default: `True`                                                                                                                     |
| `MICROSOFT_SSO_ENABLED`                          | Enable or disable the plugin. Default: `True`                                                                                                                                      |
| `MICROSOFT_SSO_GRAPH_TIMEOUT`                    | The timeout in seconds for the Microsoft Graph API requests. Default: `10`                                                                                                         |
| `MICROSOFT_SSO_LOGIN_FAILED_URL`                 | The named url path that the user will be redirected to if an authentication error is encountered. Default: `admin:index`                                                           |
| `MICROSOFT_SSO_LOGO_URL`                         | The URL of the logo to be used on the login button. Default: a bundled local logo file                                                                                             |
| `MICROSOFT_SSO_NEXT_URL`                         | The named url path that the user will be redirected if there is no next url after successful authentication. Default: `admin:index`                                                |
| `MICROSOFT_SSO_PAGES_ENABLED`                    | Enable SSO button injection on non-admin pages. Default: `None`                                                                                                                    |
| `MICROSOFT_SSO_PRE_CREATE_CALLBACK`              | Callable for processing pre-create logic. Default: `django_microsoft_sso.hooks.pre_create_user`                                                                                    |
| `MICROSOFT_SSO_PRE_CREATE_USER_RETURN_FULL_ARGS` | When `True`, the dict returned by `pre_create_callback` is used as the full kwargs for `User.objects.get_or_create()`, not just `defaults`. Default: `False`                       |
| `MICROSOFT_SSO_PRE_LOGIN_CALLBACK`               | Callable for processing pre-login logic. Default: `django_microsoft_sso.hooks.pre_login_user`                                                                                      |
| `MICROSOFT_SSO_PRE_VALIDATE_CALLBACK`            | Callable for processing pre-validate logic. Default: `django_microsoft_sso.hooks.pre_validate_user`                                                                                |
| `MICROSOFT_SSO_PROJECT_ID`                       | The Microsoft OAuth 2.0 Project ID. Default: `None`                                                                                                                                |
| `MICROSOFT_SSO_REQUIRE_SECURE_CALLBACK`          | If `True`, callback uses secure `POST` (`form_post`) and requires explicit Django `CACHES` config. If `False`, callback uses legacy `GET` query-string mode. Default: `False`      |
| `MICROSOFT_SSO_SAVE_ACCESS_TOKEN`                | Save the access token in the session. Default: `False`                                                                                                                             |
| `MICROSOFT_SSO_SAVE_BASIC_MICROSOFT_INFO`        | Save basic Microsoft info on database. Default: `True`                                                                                                                             |
| `MICROSOFT_SSO_SCOPES`                           | The Microsoft OAuth 2.0 Scopes. Default: `["User.ReadBasic.All"]`                                                                                                                  |
| `MICROSOFT_SSO_SESSION_COOKIE_AGE`               | The age of the session cookie in seconds. Default: `3600`                                                                                                                          |
| `MICROSOFT_SSO_SHOW_FAILED_LOGIN_MESSAGE`        | Show a message on browser when the user creation fails on database. Default: `False`                                                                                               |
| `MICROSOFT_SSO_STAFF_LIST`                       | List of emails that will be created as staff. Default: `[]`                                                                                                                        |
| `MICROSOFT_SSO_SUPERUSER_LIST`                   | List of emails that will be created as superuser. Default: `[]`                                                                                                                    |
| `MICROSOFT_SSO_TEXT`                             | The text to be used on the login button. Default: `Sign in with Microsoft`                                                                                                         |
| `MICROSOFT_SSO_TIMEOUT`                          | The timeout in seconds for the Microsoft SSO authentication returns info, in minutes. Default: `10`                                                                                |
| `MICROSOFT_SSO_UNIQUE_EMAIL`                     | When get or create a new user, check if the email already exists. Default: `False`                                                                                                 |
| `SSO_ADMIN_ROUTE`                                | The admin index page route. Default: `admin:index`                                                                                                                                 |
| `SSO_SHOW_FORM_ON_ADMIN_PAGE`                    | Show the form on the admin page. Default: `True`                                                                                                                                   |
| `SSO_USE_ALTERNATE_W003`                         | Use alternate W003 warning. You need to silence original templates.E003 warning. Default: `False`                                                                                  |


CHANGELOG.md
# CHANGELOG


## v7.1.1 (2026-07-27)

### Fixes

* fix: missing image in default button configuration ([`93f0d90`](https://github.com/megalus/django-microsoft-sso/commit/93f0d906d7fae755912e98e01e0b18daedb7b616))


## v7.1.0 (2026-06-17)

### Documentation

* docs: Update versions ([`5f23216`](https://github.com/megalus/django-microsoft-sso/commit/5f2321661482bae700f2b19012b51c17408763d8))

### Features

* feat: Add full control over `get_or_create` arguments in user creation

- Introduced the `MICROSOFT_SSO_PRE_CREATE_USER_RETURN_FULL_ARGS` setting to enable using the full kwargs returned by the `pre_create_callback` for `User.objects.get_or_create`.
- Updated `get_or_create_user` logic to support both default and custom user creation workflows.
- Added new test cases to ensure compatibility and correctness.
- Updated documentation to reflect changes in settings and usage examples.
- Removed outdated `.github/copilot-instructions.md`. ([`fb05693`](https://github.com/megalus/django-microsoft-sso/commit/fb05693d3f704d414f7ac9740dc37afdd0d70d0f))

### Unknown

* Merge pull request #25 from megalus/develop

Add option `MICROSOFT_SSO_PRE_CREATE_USER_RETURN_FULL_ARGS` ([`c789b11`](https://github.com/megalus/django-microsoft-sso/commit/c789b118cc7365fbede3564535a5cadd295bb8ab))

* Merge remote-tracking branch 'origin/main' ([`367e73f`](https://github.com/megalus/django-microsoft-sso/commit/367e73ffbdd020c8fad806c845edc841c7b0d726))


## v7.0.1 (2026-04-25)

### Continuous Integration

* ci: add step to restore branch context from released tag in publish workflow ([`d612c6f`](https://github.com/megalus/django-microsoft-sso/commit/d612c6f8c0e9975da7fa663653b99583bff98dcf))

* ci: add missing environment name in actions ([`5177dd6`](https://github.com/megalus/django-microsoft-sso/commit/5177dd69b52b521258ebef5509995d6b92fb816a))

### Fixes

* fix: configure dynamic versioning and wheel build settings in pyproject.toml ([`6e4b7a9`](https://github.com/megalus/django-microsoft-sso/commit/6e4b7a9cdbe3290f0a90af408a590cbc9ec38376))

### Unknown

* Merge remote-tracking branch 'origin/main' ([`d24006c`](https://github.com/megalus/django-microsoft-sso/commit/d24006ce16be87fc2acf5c45b719bf80c023334a))


## v7.0.0 (2026-04-25)

### Breaking

* feat!: implement secure callback mode with cache validation for Microsoft SSO. This is a BREAKING CHANGE commit:

* Remove Django 4.2 support
* Add Django 6.0 support
* Add Python 3.14 support
* Minimal MSAL version: 4.82.0 ([`df56880`](https://github.com/megalus/django-microsoft-sso/commit/df56880bb6d19deb294d1261e46178e8738ed329))

### Chores

* chore: move from poetry/black to uv/ruff ([`d675f2a`](https://github.com/megalus/django-microsoft-sso/commit/d675f2a6f4030a13dc906c2ce9df0a5e9ad691a8))

### Continuous Integration

* ci: update publish workflow to checkout released tag and improve pre-commit commands ([`552bff2`](https://github.com/megalus/django-microsoft-sso/commit/552bff2002af177734f38c1817805721667a4e3d))

### Features

* feat: update CI workflow to use matrix for Python and Django versions ([`de701ac`](https://github.com/megalus/django-microsoft-sso/commit/de701ac36ebdde27687877bb666bff6313c2f5ff))

* feat: expand CI matrix to include Python 3.14 and Django 6.0 ([`5421570`](https://github.com/megalus/django-microsoft-sso/commit/542157085f96fb448fb3a39a97ae688260fd3ee4))

### Fixes

* fix: update Python version in CI setup and disable build step ([`a13dd62`](https://github.com/megalus/django-microsoft-sso/commit/a13dd6277e17df7cfc98bd56824801215b30cb80))

* fix: update installation command for Django in CI workflow ([`826b051`](https://github.com/megalus/django-microsoft-sso/commit/826b0510466bdbe3fb984928779722a8e4c590ba))

* fix: msal min version (python instead dotnet) ([`e7fa343`](https://github.com/megalus/django-microsoft-sso/commit/e7fa3438cc1d677c5f0b2dc8584715aa9ef91ea3))

### Unknown

* Merge pull request #24 from megalus/feat/dual-callback-response-mode

New Version: 7 ([`897de10`](https://github.com/megalus/django-microsoft-sso/commit/897de10d2d7db18b00e2e0430e46a3ae30e8980a))

* doc: update settings.md

thanks @mauvray ([`37f7c3d`](https://github.com/megalus/django-microsoft-sso/commit/37f7c3d08c703b67da9d171d516348e02297093e))


## v6.0.2 (2025-09-16)

### Fixes

* fix: handle multiple reverse proxies in redirect URI generation ([`beb3f53`](https://github.com/megalus/django-microsoft-sso/commit/beb3f536acc254513c37c9ccbfd871211fd4520a))


## v6.0.1 (2025-09-15)

### Fixes

* fix: update code for django unfold ([`39f2014`](https://github.com/megalus/django-microsoft-sso/commit/39f201465b3965dc2d7dbd6937e8d3f320d369ef))


## v6.0.0 (2025-09-09)

### Breaking

* feat!: Version 6.0 - BREAKING CHANGE

* Add support to Django 5.2
* SSO configuration can be defined per request now
* SSO Tags can be customized in views now.
* Fix PEP 562 for django_microsoft_sso.conf
* Use Django's EMAIL_FIELD to determine email field in User ([`5e1502c`](https://github.com/megalus/django-microsoft-sso/commit/5e1502c82f73692c1addb343d2fd46efca81a218))

### Chores

* chore: additional fixes after PR ([`cdd4f53`](https://github.com/megalus/django-microsoft-sso/commit/cdd4f534e0fd6ecc535a322861aa0a20bd68ec08))

* chore: fixes from PR ([`484b675`](https://github.com/megalus/django-microsoft-sso/commit/484b6757b205385e0af37cb8b4c36ecc04f2e963))

* chore: update Makefile, pre-commit config, and docs for project setup and guidelines ([`f6f56a7`](https://github.com/megalus/django-microsoft-sso/commit/f6f56a7fecb6425461fb50bac65ef987f04c5e60))

### Continuous Integration

* ci: fix lint ([`25b51e1`](https://github.com/megalus/django-microsoft-sso/commit/25b51e1a184995ed6370293e55c9d88d798034ef))

### Documentation

* docs: add coding guidelines and project instructions ([`b689bd0`](https://github.com/megalus/django-microsoft-sso/commit/b689bd0f720f65bbb72499b77cc84852824c9d74))


example_microsoft_app/settings.py
"""
Django settings for example_microsoft_app project.

Generated by 'django-admin startproject' using Django 3.2.9.

For more information on this file, see
https://docs.djangoproject.com/en/3.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.2/ref/settings/
"""

# flake8: noqa: E731

from pathlib import Path

from loguru import logger
from stela import env

# Build paths inside the project like this: BASE_DIR / 'subdir'.

BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = "django-insecure"

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []


# Application definition

INSTALLED_APPS = [
    # Uncomment for Grappelli
    # "grappelli",
    # Uncomment for Jazzmin
    # "jazzmin",
    # Uncomment for Admin Interface
    # "admin_interface",
    # "colorfield",
    # Uncomment for Jest
    # "jet.dashboard",
    # "jet",
    # Uncomment for Unfold
    # "unfold",
    "django.contrib.admin",
    "django.contrib.auth",
    "django.contrib.contenttypes",
    "django.contrib.sessions",
    "django.contrib.messages",  # Need for Auth messages
    "django.contrib.staticfiles",
    "django.contrib.sites",  # Optional: Add Sites framework
    "django_google_sso",  # Optional: Add django_google_sso
    "django_microsoft_sso",  # Add django_microsoft_sso
]

MIDDLEWARE = [
    "django.middleware.security.SecurityMiddleware",
    "django.contrib.sites.middleware.CurrentSiteMiddleware",
    "django.contrib.sessions.middleware.SessionMiddleware",
    "django.middleware.common.CommonMiddleware",
    "django.middleware.csrf.CsrfViewMiddleware",
    "django.contrib.auth.middleware.AuthenticationMiddleware",
    "django.contrib.messages.middleware.MessageMiddleware",
    "django.middleware.clickjacking.XFrameOptionsMiddleware",
]

ROOT_URLCONF = "example_microsoft_app.urls"

TEMPLATES = [
    {
        "BACKEND": "django.template.backends.django.DjangoTemplates",
        "DIRS": [
            BASE_DIR / "example_microsoft_app" / "templates",
        ],
        "NAME": "default",
        "APP_DIRS": True,
        "OPTIONS": {
            "context_processors": [
                "django.template.context_processors.debug",
                "django.template.context_processors.request",
                "django.contrib.auth.context_processors.auth",
                "django.contrib.messages.context_processors.messages",
            ],
        },
    },
]

WSGI_APPLICATION = "example_microsoft_app.wsgi.application"


# Database
# https://docs.djangoproject.com/en/3.2/ref/settings/#databases

DATABASES = {
    "default": {
        "ENGINE": "django.db.backends.sqlite3",
        "NAME": BASE_DIR / "db.sqlite3",
    }
}


# Password validation
# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.CommonPasswordValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.NumericPasswordValidator",
    },
]


# Internationalization
# https://docs.djangoproject.com/en/3.2/topics/i18n/

LANGUAGE_CODE = "en-us"

TIME_ZONE = "UTC"

USE_I18N = True

USE_L10N = True

USE_TZ = True

if "jet" in INSTALLED_APPS:
    # Jet Theme
    JET_DEFAULT_THEME = "light-gray"

    JET_THEMES = [
        {
            "theme": "default",  # theme folder name
            "color": "#47bac1",  # color of the theme's button in user menu
            "title": "Default",  # theme title
        },
        {"theme": "green", "color": "#44b78b", "title": "Green"},
        {"theme": "light-green", "color": "#2faa60", "title": "Light Green"},
        {"theme": "light-violet", "color": "#a464c4", "title": "Light Violet"},
        {"theme": "light-blue", "color": "#5EADDE", "title": "Light Blue"},
        {"theme": "light-gray", "color": "#222", "title": "Light Gray"},
    ]


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.2/howto/static-files/

STATIC_URL = "/static/"
STATICFILES_DIRS = [BASE_DIR / "example_microsoft_app" / "static"]


# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"

AUTHENTICATION_BACKENDS = [
    "django.contrib.auth.backends.ModelBackend",
    "example_microsoft_app.backend.MyBackend",
]


SITE_ID = 1

# Cache configuration is needed when MICROSOFT_SSO_REQUIRE_SECURE_CALLBACK = True
CACHES = {
    "default": {
        "BACKEND": "django.core.cache.backends.locmem.LocMemCache",
        "LOCATION": "unique-snowflake",
    }
}

###############################
#                             #
# Test Microsoft              #
#                             #
###############################

# Uncomment MICROSOFT_SSO_CALLBACK_DOMAIN to use Sites Framework site domain
# Or comment both and use domain retrieved from accounts/login/ request
MICROSOFT_SSO_CALLBACK_DOMAIN = env.MICROSOFT_SSO_CALLBACK_DOMAIN

#####################
# Callback Examples #
#####################


def get_application_id(request):
    return env.MICROSOFT_SSO_APPLICATION_ID


MICROSOFT_SSO_APPLICATION_ID = get_application_id  # Dynamic
MICROSOFT_SSO_CLIENT_SECRET = env.MICROSOFT_SSO_CLIENT_SECRET  # Static

###########################
# Admin and Page Examples #
###########################

# --8<-- [start:sso_config]
# settings.py
from django_microsoft_sso.helpers import is_admin_path  # noqa: E402


def get_sso_config(request):
    config = {
        "admin": {
            "allowable_domains": env.get_or_default("MICROSOFT_SSO_ALLOWABLE_DOMAINS", []),
            "login_failed_url": "admin:login",
            "next_url": "admin:index",
            "session_cookie_age": 3600,  # 1 hour - default
            "staff_list": env.get_or_default("MICROSOFT_SSO_STAFF_LIST", []),
            "superuser_list": env.get_or_default("MICROSOFT_SSO_SUPERUSER_LIST", []),
            "auto_create_first_superuser": True,  # Create superuser on first eligible user login
        },
        "pages": {
            "allowable_domains": ["*"],  # Allow all domains
            "login_failed_url": "index",
            "next_url": "secret",
            "session_cookie_age": 86400,  # 24 hours
            "staff_list": [],
            "superuser_list": [],
            "auto_create_first_superuser": False,
        },
    }
    if is_admin_path(request):
        logger.debug("Returning Admin SSO configuration")
        return config["admin"]
    else:
        logger.debug("Returning Pages SSO configuration")
        return config["pages"]


# Configure settings as callables
MICROSOFT_SSO_ALLOWABLE_DOMAINS = lambda request: get_sso_config(request)[
    "allowable_domains"
]
MICROSOFT_SSO_AUTO_CREATE_FIRST_SUPERUSER = lambda request: get_sso_config(request)[
    "auto_create_first_superuser"
]
MICROSOFT_SSO_STAFF_LIST = lambda request: get_sso_config(request)["staff_list"]
MICROSOFT_SSO_SUPERUSER_LIST = lambda request: get_sso_config(request)["superuser_list"]
MICROSOFT_SSO_NEXT_URL = lambda request: get_sso_config(request)["next_url"]
MICROSOFT_SSO_SESSION_COOKIE_AGE = lambda request: get_sso_config(request)[
    "session_cookie_age"
]
MICROSOFT_SSO_LOGIN_FAILED_URL = lambda request: get_sso_config(request)["login_failed_url"]
# --8<-- [end:sso_config]

#########################################
# Split SSO between Admin/Page Examples #
#########################################

# --8<-- [start:sso_split]
# settings.py

# Control globally - both Admin and Pages (default: True)
MICROSOFT_SSO_ENABLED = True
GOOGLE_SSO_ENABLED = True

# Use Google SSO for Pages only
# Always define both Admin and Pages settings
GOOGLE_SSO_PAGES_ENABLED = True
GOOGLE_SSO_ADMIN_ENABLED = False

# Explicitly configure Google Settings
# to make sure they cannot have Admin privileges
GOOGLE_SSO_ALLOWABLE_DOMAINS = ["*"]
GOOGLE_SSO_AUTO_CREATE_FIRST_SUPERUSER = False
GOOGLE_SSO_STAFF_LIST = []
GOOGLE_SSO_SUPERUSER_LIST = []
GOOGLE_SSO_FAILED_URL = "index"
GOOGLE_SSO_NEXT_URL = "secret"
# --8<-- [end:sso_split]

# Optional: You can save access token to session
MICROSOFT_SSO_SAVE_ACCESS_TOKEN = True

# Optional: Start or Stop User auto-creation
# MICROSOFT_SSO_AUTO_CREATE_USERS = True

# Optional: Show failed login attempt message on browser.
# This message can be used in exploit attempts.
# MICROSOFT_SSO_SHOW_FAILED_LOGIN_MESSAGE = False

# Optional: Add if you want to use custom authentication backend
MICROSOFT_SSO_AUTHENTICATION_BACKEND = "example_microsoft_app.backend.MyBackend"

# Optional: Change Scopes
MICROSOFT_SSO_SCOPES = [
    # "User.ReadBasic.All"  # default scope
    "User.Read.All",  # additional scope
]

# Optional: Add pre-validate logic
MICROSOFT_SSO_PRE_VALIDATE_CALLBACK = "backend.pre_validate_callback"

# Optional: Add pre-create logic
MICROSOFT_SSO_PRE_CREATE_CALLBACK = "backend.pre_create_callback"

# Optional: Add pre-login logic
# MICROSOFT_SSO_PRE_LOGIN_CALLBACK = "backend.pre_login_callback"

# Optional: Always update user data
MICROSOFT_SSO_ALWAYS_UPDATE_USER_DATA = True

# Optional: Require secure callback for form_post response mode
# You must set the CACHES setting to use this feature.
MICROSOFT_SSO_REQUIRE_SECURE_CALLBACK = True  # False=GET, True=POST

# Optional: Customize Button Text
# MICROSOFT_SSO_TEXT = "Login using Microsoft 365 Account"

# Optional: Disable Logs
# MICROSOFT_SSO_ENABLE_LOGS = False

# Optional: Disable Django Messages
# MICROSOFT_SSO_ENABLE_MESSAGES = False

###############################
#                             #
# Test Google                 #
#                             #
###############################

# Uncomment GOOGLE_SSO_CALLBACK_DOMAIN to use Sites Framework site domain
# Or comment both and use domain retrieved from accounts/login/ request
GOOGLE_SSO_CALLBACK_DOMAIN = env.GOOGLE_SSO_CALLBACK_DOMAIN

GOOGLE_SSO_SESSION_COOKIE_AGE = 3600  # default value
GOOGLE_SSO_CLIENT_ID = env.GOOGLE_SSO_CLIENT_ID
GOOGLE_SSO_PROJECT_ID = env.GOOGLE_SSO_PROJECT_ID
GOOGLE_SSO_CLIENT_SECRET = env.GOOGLE_SSO_CLIENT_SECRET

GOOGLE_SSO_TIMEOUT = 10  # default value
GOOGLE_SSO_SCOPES = [  # default values
    "openid",
    "https://www.googleapis.com/auth/userinfo.email",
    "https://www.googleapis.com/auth/userinfo.profile",
    # "https://www.googleapis.com/auth/user.birthday.read",  # additional scope
]

# Optional: Add if you want to use custom authentication backend
GOOGLE_SSO_AUTHENTICATION_BACKEND = "example_microsoft_app.backend.MyBackend"

# Optional: You can save access token to session
# GOOGLE_SSO_SAVE_ACCESS_TOKEN = True

# Optional: Add pre-login logic
# GOOGLE_SSO_PRE_LOGIN_CALLBACK = "backend.pre_login_callback"

GOOGLE_SSO_LOGO_URL = (
    "https://upload.wikimedia.org/wikipedia/commons/thumb/c/c1/"
    "Google_%22G%22_logo.svg/1280px-Google_%22G%22_logo.svg.png"
)

# Uncomment to disable user auto-creation
# GOOGLE_SSO_AUTO_CREATE_USERS = False  # default: True

# Uncomment to hide login form on admin page
# SSO_SHOW_FORM_ON_ADMIN_PAGE = False  # default: True

# If you're using more than one SSO provider,
# you can full disable django's E003/W003 check, uncomment the
# SILENCED_SYSTEM_CHECKS option
# To use an alternate version for this check,
# which filters SSO templates, uncomment both options:
SILENCED_SYSTEM_CHECKS = ["templates.W003"]
SSO_USE_ALTERNATE_W003 = True  # default: False

# Optional: silence secure callback cache warning when expected in your setup
# SILENCED_SYSTEM_CHECKS += ["sso.W001"]
